terminal/TerminalServer.cpp

```diff
@@ -59,18 +63,13 @@ void runTerminal(shared_ptr<ServerClientConnection> serverClientState,
     // Data structures needed for select() and
     // non-blocking I/O.
     fd_set rfd;
-    fd_set wfd;
-    fd_set efd;
     timeval tv;
 
     FD_ZERO(&rfd);
-    FD_ZERO(&wfd);
-    FD_ZERO(&efd);
     FD_SET(masterfd, &rfd);
-    FD_SET(STDIN_FILENO, &rfd);
```

Comment:
`etserver` never actually *reads* from fd 0, at least that I saw in strace or in your source.  (I wouldn't expect a network daemon running in background to have anything usefully connected on the standard file descriptors.)  It checks for console output from the user's session on `masterfd` here, with a timeout of 1ms, and then reads it in the block just below here.  It checks for network input (with a *different* `select()`) and reads it in `UnixSocketHandler::hasData()` with a timeout of 0.

With the original code, it's also checking for input on fd 0.  'select()` never waits and always returns a ready status from that file descriptor.  I didn't delve into why, but probably fd 0 is closed or otherwise returning an error, which will cause `select()` to report that fd as readable.  So the program consumes 100% CPU.

Removing this reference makes the program repeatedly sleep for 1ms in the first `select()` if nothing is available, polling the other `select()` each time.  This is still inefficient code, but far better than before.  This patch applies that simple band-aid.

Ideally you would use a single `select()` for the entire program and not have to time out so you can go poll some other `select()` and fds.  That, or convert the program to use an event handling framework.